installer/pkg/config-generator/generator.go

```diff
@@ -62,8 +66,93 @@ func New(cluster config.Cluster) ConfigGenerator {
 	}
 }
 
+// maoOperatorConfig contains configuration for mao managed stack
+// TODO(enxebre): move up to "github.com/coreos/tectonic-config
+type maoOperatorConfig struct {
+	metav1.TypeMeta `json:",inline"`
+	TargetNamespace string         `json:"targetNamespace"`
+	APIServiceCA    string         `json:"apiServiceCA"`
+	Provider        string         `json:"provider"`
+	AWS             *awsConfig     `json:"aws"`
+	Libvirt         *libvirtConfig `json:"libvirt"`
+}
+
+type libvirtConfig struct {
+	ClusterName string `json:"clusterName"`
+	URI         string `json:"uri"`
+	NetworkName string `json:"networkName"`
+	IPRange     string `json:"iprange"`
+	Replicas    int    `json:"replicas"`
+}
+
+type awsConfig struct {
+	ClusterName      string `json:"clusterName"`
+	ClusterID        string `json:"clusterID"`
+	Region           string `json:"region"`
+	AvailabilityZone string `json:"availabilityZone"`
+	Image            string `json:"image"`
+	Replicas         int    `json:"replicas"`
+}
+
+func (c *ConfigGenerator) maoConfig(clusterDir string) (*maoOperatorConfig, error) {
+	cfg := maoOperatorConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "machineAPIOperatorConfig",
+		},
+
+		TargetNamespace: maoTargetNamespace,
+	}
+
+	ca, err := ioutil.ReadFile(filepath.Join(clusterDir, aggregatorCACertPath))
+	if err != nil {
+		return nil, fmt.Errorf("Could not read aggregator CA: %v", err)
```

Comment:
Error strings [should not be capitalized][1].  I thought `golint` would pick this up, but I guess not :p.

[1]: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
Fixed here. It looks like there are tons of instances of this.

Reply:
> Fixed here. It looks like there are tons of instances of this.

I'm fine as long as we aren't adding new ones ;).  We can chip away at existing instances in follow-up work.  Looks like you're still adding some in other places:

```console
$ git diff origin/master..origin/pr/119 -- installer | grep '^diff\|^+.*[Ee]rror.*"[A-Z]' | grep -B1 '[Ee]rror'
diff --git a/installer/pkg/config-generator/generator.go b/installer/pkg/config-generator/generator.go
+		return nil, fmt.Errorf("Unknown provider for machine-api-operator: %v", cfg.Provider)
--
diff --git a/installer/pkg/workflow/destroy.go b/installer/pkg/workflow/destroy.go
+		return fmt.Errorf("Failed to build cluster-api client: %v", err)
+		return fmt.Errorf("Failed to scale worker MachineSet: %v", err)
+		return fmt.Errorf("Worker MachineSet failed to scale down: %v", err)
+		return fmt.Errorf("Failed to delete worker MachineSet: %v", err)
+		return nil, fmt.Errorf("Failed to build config: %v", err)
+		return nil, fmt.Errorf("Failed to build client: %v", err)
```

Reply:
> I thought golint would pick this up, but I guess not :p.

Ah, this is because error-cap violations have a confidence of 0.6][1] (vs. [the default 0.8][2].  Running with:

```console
$ golint -min_confidence 0.6 ./installer/...
```

turns them up.

[1]: https://github.com/golang/lint/blob/06c8688daad7faa9da5a0c2f163a3d14aac986ca/lint.go#L1186
[2]: https://github.com/golang/lint/blob/06c8688daad7faa9da5a0c2f163a3d14aac986ca/golint/golint.go#L23

Reply:
Fixed.